pages/Graph/Overview.md

```diff
@@ -16,13 +21,86 @@ This section should also cover how our graph representation could be applied to
+
+## Graph Storage
+All graph change events are signed and added to a [batch file](/Batches/Overview).
+Once that batch file is complete, it is [announced](/Messages/Announce) on the blockchain per the DSNP specification. 
+Each announce event specifies the type of ([dsnpType](/Messages/Announce)) messages that are in the batch file. 
+
+## Graph Retrieval, Ordering & Reading
+Each graph change event reflects a change in state for a given user's graph (i.e. when a Bob follows Charlie, that "follow" is then reflected as a state change in Bob's graph).
+As mentioned above, graph change events get added to a batch file which gets signed and then announced on chain. 
+
+A user's graph can be derived for any given point in time by retrieving the graph change events for that time period (i.e. from block 0 to block 10). 
+Once those graph change events are retrieved, they can be ordered (as mentioned below) to reflect the current graph state
+(i.e. Charlie has followed Bob then unfollowed him and then followed him again. The graph state reflects that Charlie is Following Bob).
+
+To retrieve the graph, do the following:
+1. Retrieve the log events[dsnpType](/Messages/Types)`GraphChange` from the chain for a point in time (i.e. from a specific block number to a specific block number)
+1. Retrieve the batch file from each log event. Each log event of type [GraphChange](/Messages/Announce) has a field called `dsnpUri` which contains a uri pointing to a [batch file](/Batches/Overview). 
+1. Query the batch files for the data for a particular DSNP User Id to retrieve information about the respective graph. For more on how batch file storage and how to query the batch file see - [batches overview](/Batches/Overview)
+1. Order the retrieved data based on the following  
+    1. Block Number Ascending
+    1. Transaction Index Ascending
+    1. Log Index Ascending
+    1. Batch File Row Index Ascending
+                                                
+For more on ordering see [Message Ordering Specification](/Messages/Ordering).
+
+#### Creating the graph
+Each graph change event represents a state transition for the graph. 
+The state of the graph at any time is given by taking the state of the graph at a previous time and applying all graph change events not previously applied in the order specified above.
+
+This can be done by retrieving `GraphChange` log events starting with a specific block number. 
+For example: The graph for a given user has been retrieved and stored up until block number 18. 
+A filter can be added to the logs being retrieved to only retrieve log events starting from block 19.
 
-TBD
+## Replay Attacks
 
-## Details
+[Replay attacks](https://en.wikipedia.org/wiki/Replay_attack) are prevented by ensuring that each graph change event is identifiably unique. 
```

Comment:
This is passive. I would rephrase:
Clients must ignore any GraphChange event that comes after another event with the same signature. This avoids [Replay attacks](https://en.wikipedia.org/wiki/Replay_attack).